backend/question-service/src/controllers/question.controller.ts

```diff
@@ -171,3 +173,28 @@ export async function handleGetCategories(request: Request, response: Response):
     }
     response.status(200).json(categories).send()
 }
+
+export async function handleGetRandomQuestion(request: Request, response: Response): Promise<void> {
+    const topic = request.query.topic
+    const complexity = request.query.complexity
+    if (!topic || !complexity) {
+        response.status(400).json('complexity or topic missing').send()
+        return
+    }
+    if (!Object.values(Category).includes(topic as Category)) {
+        console.log(topic, complexity, Object.values(Category).includes(topic as Category))
+        response.status(400).json('invalid topic').send()
+        return
+    }
+    if (!Object.values(Complexity).includes(complexity as Complexity)) {
+        response.status(400).json('invalid complexity').send()
+        return
+    }
```

Comment:
what happens if we directly perform a query with the given params and just throw an error (404 if not found)?

Reply:
it will throw an error (404Not Found), i would think this error handling will make it clearer as to which error we are facing